shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java

```diff
@@ -108,15 +106,6 @@
      * @return subquery where segment collection.
      */
     public Collection<WhereSegment> getSubqueryWhereSegments() {
-        Collection<WhereSegment> subqueryWhereSegments = new ArrayList<>();
```

Comment:
Hi, Sorry for the late feedback.
Do you think it is possible to move this function to `SelectStatementContext` referring to [#6319](https://github.com/apache/shardingsphere/pull/6319/files).
Basically, `xxxStatement` is designed as a POJO object, and we put more logic coding into `xxxStatementContext` for `rewriting`.

Reply:
Besides, I presume #6319 finished part of your work, so my suggestion is you'd better give it check, and if possible, a coding merging and optimizing is welcomed. :-)

Reply:
@tristaZero Thank you for your suggestions, I will check and optimize this function. 😊